### PR TITLE
[dart-jaguar] Adds option to customise the generated API client name

### PR DESCRIPTION
**Pull request by athornz**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [X] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Derives the generated API class name from the pub name, rather than the generic `JaguarApiGen` name.

CC @ircecho @swipesight @jaumard

### CODE REVIEW

**Comment by wing328** [comment]
@athornz thanks for the PR. Quick question about the client name. Is that related to the pubName setting somehow?
```
        pubName
            Name in generated pubspec
```

If yes, shall we reuse `pubName` instead?

(disclosure: no expert in Dart)

**Comment by athornz** [comment]
It could make sense to reuse it, but the naming convention for dart packages is snake_case which is likely not what the user would want for the class name.

**Comment by wing328** [comment]
> dart packages is snake_case which is likely not what the user would want for the class name.

We can use camelize, etc string utility function to convert `pubName` into proper format for {{clientName}} in the generator

**Comment by athornz** [comment]
Sure that seems like a good option 👍

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartJaguarClientCodegen.java [hunk 1]
```diff
@@ -142,6 +147,13 @@ public void processOpts() {
             additionalProperties.put(PUB_NAME, pubName);
         }
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartJaguarClientCodegen.java]: @athornz what about setting clientName here using StringUtils methods?

That way we don't need to introduce another option

Comment (athornz) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartJaguarClientCodegen.java]: Great suggestion, works really well. I've pushed the changes now.

**Comment by wing328** [comment]
The CircleCI error is not related to this PR.

Since it's a breaking change with fallback, I've retargeted this PR to 4.1.x branch.

**Comment by wing328** [comment]
Looks like the rebase has failed and therefore this PR contains commits not authored by you.

I'll try to cherry-pick your commits and create a new PR.

**Comment by athornz** [comment]
Thanks!

**Comment by wing328** [comment]
@athornz have a nice weekend

**Comment by wing328** [comment]
@athornz thanks for the PR, which has been included in the 4.1.0 release: https://twitter.com/oas_generator/status/1160000504455319553

**Pull request merged**